### ISSUE TITLE
Quoted TypeAlias annotation is not treated as type alias

### ISSUE THREAD

**Issue by killjoy1221** [microsoft/pyright#3934]
**Describe the bug**

When using `typing.TypeAlias` or `typing_extensions.TypeAlias` with quotes, the variable fails to be treated as a type alias. It is instead treated as a normal variable of type `TypeAlias`.

This makes it difficult to develop for older python versions without TypeAlias without adding `typing_extensions` as a full dependency or declaring all the type aliases inside `if TYPE_CHECKING:`.

**To Reproduce**
Create a type alias with `TypeAlias`. Wrap the annotation in quotes to make it a ForwardRef.

Type checking will cause the following error
```
Expression of type "{}" cannot be assigned to declared type "TypeAlias"
  "{} is incompatible with "TypeAlias"
```

**Expected behavior**
Pyright treats the variable as a type alias

**Screenshots or Code**
```py
from typing_extensions import TypeAlias, reveal_type

string1: "TypeAlias" = "str" # "Literal['str']" is incompatible with "TypeAlias"
string2: TypeAlias = "str"

reveal_type(string1) # Type of "string1" is "TypeAlias"
reveal_type(string2) # Type of "string2" is "Type[str]"

```

**VS Code extension or command-line**
VSCode extension
ms-python.python: v2022.14.0
ms-python.vscode-pylance: v2022.9.10

**Comment by erictraut** [comment]
`TypeAlias` requires very special processing in pyright because it's not really a type annotation. In any other language, it would have its own statement with a dedicated keyword. In fact, I've authored a draft [PEP 695](https://peps.python.org/pep-0695/#generic-type-alias) that proposes to add such a statement to Python 3.12.

I'll see if I can easily extend pyright's binder (the stage where `TypeAlias` is processed) to accommodate a quoted form.

If that's not possible, another workaround that you can use is `from __future__ import annotations`, which effectively treats all type annotations as though they are quoted.

Out of curiosity, why you are reluctant to take a dependency on `typing_extensions`? It's a small library with little or no runtime overhead. Adding `if TYPE_CHECKING` checks and quotes to avoid it sounds cumbersome and difficult to maintain by comparison.

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.271, which I just published. It will also be included in a future release of pylance.
